File: argus/framework-template-python/tests/setup/auth_setup.py
"""Auth-once helper: log in through the UI a single time and persist storage_state.

The conftest `storage_state` session fixture calls this once per run (re-authenticating
and overwriting the saved state each invocation); every UI test then starts already
authenticated (the canonical Playwright storageState pattern — no per-test login). This
module is import-path-independent on purpose: conftest adds
``tests/setup`` to sys.path and imports it lazily, so it never affects API-only runs.
"""
from __future__ import annotations

import re
from pathlib import Path

from playwright.sync_api import Browser

from qa.config import ENV
from qa.pages.login_page import LoginPage

# <-- adapt the post-login success signal to the app under test.
SUCCESS_URL = re.compile(r"dashboard|home|/$")


def authenticate(browser: Browser, role: str, state_path: str) -> None:
    """Drive a UI login for `role` and write the resulting storage_state to `state_path`."""
    Path(state_path).parent.mkdir(parents=True, exist_ok=True)
    context = browser.new_context(base_url=ENV.ui_url)
    page = context.new_page()
    try:
        LoginPage(page).login_as(role)
        page.wait_for_url(SUCCESS_URL)
        context.storage_state(path=state_path)
    finally:
        context.close()

File: argus/framework-template/tests/setup/auth.setup.ts
import { test as setup } from '@playwright/test';
import { LoginPage } from '../../src/pages/login.page';

// Canonical Playwright UI-auth: log in ONCE here, persist storageState,
// every test in the `ui` project starts already authenticated (see playwright.config.ts).
const USER_STATE = `${process.env.ARGUS_AUTH_DIRECTORY ?? '.auth'}/user.json`;

setup('authenticate as user via UI', async ({ page }) => {
  const loginPage = new LoginPage(page);
  await loginPage.loginAs('user');
  await page.waitForURL(/dashboard|home|\/$/); // <-- adapt the success signal
  await page.context().storageState({ path: USER_STATE });
});
